### DISCUSSION TITLE
How to use globally declared enums, interfaces, or types in nuxt3?

### DISCUSSION THREAD

**Discussion by FatNerdPeng** [nuxt/nuxt#25930] [Questions]
I have a function in the `.vue` file in the `pages` directory:
```
<script setup lang=-""ts>
...
async function getUser(): Promise<void> {
  try {
    const { data, pending, error, refresh } = await useFetchLocal(
      '/user/info',
      RequestMethod.Get
    );
  } catch (error) {
    console.log(' ~ getUser ~ error:', error);
  }
}
...
</script>
```
This `RequestMethod` is an enumeration declared by me in a file called `common. d. ts` in the `utils` directory:
```
declare enum RequestMethod {
  Get = 'get',
  Post = 'post',
  Put = 'put',
  Patch = 'patch',
  Delete = 'delete',
}
```
and then it reports an error:
`ReferenceError: RequestMethod is not defined`
I am a noob in TypeScript and Nuxt, so I don't know where I made a mistake, but it didn't prompt me for an error during compilation, so this is a runtime error?

**Comment by FrazeColder** [comment]
@FatNerdPeng did you figure out how to do that?

**Reply by dsvgl** [reply]
Enums are both, Runtime code and types and should not be put inside a `d.ts` file.

**Comment by nowo** [comment]
I like to add a `config` folder in a project to store global things, and then add `imports: {dirs: ['config']}` to the `nuxt. config. ts` file. The content of the config file will be automatically imported globally, for example:

![image](https://github.com/nuxt/nuxt/assets/70360040/9a4a4fc1-d865-4109-b98b-114a00b2e442)

Note: The `enum` type must use `export`, which is required at runtime

https://nuxt.com/docs/api/nuxt-config#imports
